src/lib/actions/AbstractRunAction.ts

```diff
@@ -90,11 +90,13 @@ export abstract class AbstractRunAction implements Action {
 		const resultsProcessor: ResultsProcessor = this.resultsProcessorFactory.createResultsProcessor(
 			this.display, outputOptions, jsonReturnValueHolder);
 
+		const primaryProjectDir: string = this.inputProcessor.resolveProjectDirPaths(inputs)[0];
```

Comment:
This was the cleanest way to get the projectDir all the way into where we want it. Sadly the RuleManager isn't constructed (but is pulled out of thin air) and so sadly, we need to extract out what we want and pass it through all the functions. Thankfully I could just make this an optional parameter so we don't need to update all the tests